src/main/java/com/microsoft/graph/authentication/IAuthenticationProvider.java

```diff
@@ -0,0 +1,15 @@
+package com.microsoft.graph.authentication;
+
+import com.microsoft.graph.exceptions.AuthenticationException;
+import com.microsoft.graph.httpcore.IHttpRequest;
+
+public interface IAuthenticationProvider {
```

Comment:
so this and ICoreAuthenticationProvider have been moved to this package from httpcore but we haven't removed the httpcore files.  I assume that's for some sort of back compat, correct?  I think it might be worth doing a breaking change to clean this interface mess up and get rid of ICoreAuthenticationProvider.

Reply:
Correct, I also initially meant to place IHttpRequest in a new folder called just Http which is the namespace in the service library. I think with HttpCore we might consider both renaming as well as cleaning up the location of several of the files in that folder.